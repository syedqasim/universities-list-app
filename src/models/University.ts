// University.ts
class University {
  name: string;
  country: string;
  web_pages: string[];
  domains: string[];

  constructor(
    name: string,
    country: string,
    web_pages: string[],
    domains: string[]
  ) {
    this.name = name;
    this.country = country;
    this.web_pages = web_pages;
    this.domains = domains;
  }
}

export default University;
