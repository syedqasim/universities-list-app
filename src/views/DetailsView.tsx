import React from "react";
import { useLocation } from "react-router-dom";
import Details from "../components/Details";
import University from "../models/University";

const DetailsView: React.FC = () => {
  const location = useLocation();
  const { university } = location.state as { university: University };

  return <Details university={university} />;
};

export default DetailsView;
