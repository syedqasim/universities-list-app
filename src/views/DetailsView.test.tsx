import React from "react";
import { render, fireEvent } from "@testing-library/react";
import { MemoryRouter, Route, Routes } from "react-router-dom";
import DetailsView from "./DetailsView";
import University from "../models/University";

const mockUniversity: University = {
  name: "Mock University",
  country: "Mock Country",
  web_pages: ["http://www.mockuniversity.com"],
  domains: ["mockuniversity.com"],
};

describe("DetailsView", () => {
  it("renders correctly and matches snapshot", () => {
    const { asFragment } = render(
      <MemoryRouter
        initialEntries={[
          { pathname: "/details", state: { university: mockUniversity } },
        ]}
      >
        <Routes>
          <Route path="/details" element={<DetailsView />} />
        </Routes>
      </MemoryRouter>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
