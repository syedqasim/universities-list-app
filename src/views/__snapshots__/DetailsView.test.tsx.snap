// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailsView renders correctly and matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="details-container"
  >
    <h1
      class="details-title"
    >
      Mock University
    </h1>
    <table
      class="details-table"
    >
      <tbody>
        <tr>
          <th>
            Country
          </th>
          <td>
            Mock Country
          </td>
        </tr>
        <tr>
          <th>
            Web Pages
          </th>
          <td>
            http://www.mockuniversity.com
          </td>
        </tr>
        <tr>
          <th>
            Domains
          </th>
          <td>
            mockuniversity.com
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</DocumentFragment>
`;
